src/models/chat/index.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * The action types accepted by the Analytics Events API.
+ *
+ * @public
+ */
+export const ACTIONS = [
+  'ADD_TO_CART',
+  'ALL_TAB_NAVIGATION',
+  'APPLY',
+  'AUTO_COMPLETE_SELECTION',
+  'BACKWARD_PAGINATE',
+  'BOOK',
+  'BRAND_ICON',
+  'CALL_TO_ACTION',
+  'CASE_START',
+  'CASE_SUBMITTED',
+  'CHAT_IMPRESSION',
+  'CHAT_LINK_CLICK',
+  'CHAT_RESPONSE',
+  'COLLAPSE',
+  'DIRECTIONS',
+  'EVENT',
+  'EXPAND',
+  'FEATURED_MESSAGE',
+  'FILTERING_WITHIN_SECTION',
+  'FORWARD_PAGINATE',
+  'HEADER_LINKS',
+  'ITEM_IN_LIST',
+  'MAP_CARD',
+  'MAP_PIN',
+  'MENU',
+  'MESSAGE',
+  'ORDER',
+  'PAGINATE',
+  'PHONE',
+  'POST',
+  'PRESET_PROMPT',
+  'PRODUCT',
+  'PROFILE',
+  'QUESTION_FOCUS',
+  'QUESTION_SUBMIT',
+  'REMOVED_FILTER',
+  'REVIEW',
+  'SCROLL_TO_BOTTOM_OF_PAGE',
+  'SEARCH_BAR_IMPRESSION',
+  'SEARCH_CLEAR_BUTTON',
+  'THUMBS_DOWN',
+  'THUMBS_UP',
+  'TICKET_URL',
+  'TITLE',
+  'VERTICAL_TAB_NAVIGATION',
+  'VERTICAL_VIEW_ALL',
+  'VOICE_START',
+  'VOICE_STOP',
+  'WEBSITE',
+] as const;
+
+/**
+ * The action types accepted by the Analytics Events API.
+ *
+ * @public
+ */
+export type Action = (typeof ACTIONS)[number];
+
+/**
+ * The payload accepted by the Analytics Events API.
+ *
+ * @public
+ */
+export interface EventPayload {
+  action: Action;
+  sessionId?: string;
+  pageUrl?: string;
+  destinationUrl?: string;
+  label?: string;
+  locale?: string;
+  timestamp?: string;
+  bot?: string; // Whether the event is from a bot
+  browserAgent?: {
+    browser?: string;
+    browserVersion?: string;
+    os?: string;
+    osVersion?: string;
+    device?: string;
+    deviceClass?: string;
+    userAgent?: string;
+  };
+  clientSdk?: Record<string, string>;
+  count?: number;
+  customTags?: Record<string, string>;
+  customValues?: Record<string, number>;
+  chat?: ChatDomainProperties;
+  entity?:
+    | {
+        entityId: string;
+      }
+    | {
+        entityUid: number;
+      };
+  ip?: {
+    address: string;
+    algorithm?: string;
+  };
+}
+
+/**
+ * Analytics event properties specific to Yext Chat.
+ *
+ * @public
+ */
+export interface ChatDomainProperties {
+  botId: string;
+  conversationId?: string;
+  sessionId?: string;
```

Comment:
Session Id is currently not a property of the chat domain. We're also missing responseId.  

Current chat domain looks like https://gerrit.yext.com/plugins/gitiles/alpha/+/master/src/com/yext/analytics/events/resources/analytics-event-v1.json#278

Reply:
How would one register a sessions start for chat?

Reply:
Yeah good call, this is a mistake. Also @AjayBenno - should we add the pages domain as well? A chat analytics event can take place on a Yext Page, after all.

@benmcginnis - right now you will have to do this manually, but at some point soon (perhaps in this pull request) we should add a method to this SDK for `startSession` and `getSession`, to standardize our approach to this.

Reply:
We had explicitly not accepted pages domain data yet since M1 was just for chat. If we need to have pages be accepted we'll need to update our endpoint to be able to start accepting data for it.